usecase/src/executor.rs

```diff
@@ -0,0 +1,27 @@
+use std::{future::Future, sync::OnceLock};
+
+use tokio::{
+    sync::Mutex,
+    task::{AbortHandle, JoinSet},
+};
+
+static EXECUTOR: Mutex<OnceLock<JoinSet<()>>> = Mutex::const_new(OnceLock::new());
+
+pub fn spawn<T, F: Future<Output = T> + Send + 'static>(future: F) -> AbortHandle {
+    tokio::task::block_in_place(|| {
+        let mut executor_guard = EXECUTOR.blocking_lock();
+        let _ = executor_guard.get_or_init(JoinSet::new);
+        let executor =
+            executor_guard.get_mut().expect("executor join set to be initialized upon access");
+        executor.spawn(async move {
+            let _ = future.await;
+        })
+    })
+}
```

Comment:
block_in_place doesn't seem to be used right here. It temporarily converts the current asynchronous task into a blocking one, ifaik
wdyt of this instead?
```suggestion
pub async fn spawn<T, F: Future<Output = T> + Send + 'static>(future: F) -> AbortHandle {
    let mut executor_guard = EXECUTOR.lock().await;
    let executor = executor_guard.get_or_init(JoinSet::new);
    executor.spawn(async move {
        let _ = future.await;
    })
}
```

Reply:
1) won't work as i'd like to preserve the signature of spawn being a synchronous function
2) to use blocking_lock we must enclose the spawning in `block_in_place` or `spawn_blocking` or `block_on`. IMO this is not an issue since this spawn is extremely fast